packages/core/bootstrap/src/lib/ws/epics.ts

```diff
@@ -154,16 +162,22 @@ export const connectEpic: Epic<AnyAction, AnyAction, any, any> = (action$, state
       const messages$ = action$.pipe(
         filter(messageReceived.match),
         filter((action) => wsHandler.filter(action.payload.message)),
+        withLatestFrom(state$),
+        map(([action, state]) => {
+          const input = state.ws.subscriptions[action.payload.subscriptionKey]?.input || {}
+          if (!input) logger.warn(`WS: Could not find subscription from incoming message`)
+          return { ...action, input }
+        }),
```

Comment:
I'd prefer we don't add to the action here, but we remove this whole map operator and move the logic to the `mergeMap` below.